images/scripts/virt-install-rhel4edge

```diff
@@ -56,44 +58,52 @@ cleanup_builder_vm () {
   fi
 }
 
-# Using virt-install is fundamentally an interactive process.  We
-# expect the installation to finish unattended, but it might drop into
-# an emergency shell upon unexpected errors, for example.  Thus, we
-# run it under 'expect' and catch a few known scenarios that require
-# special handling.
-#
-# (Also, virt-install works best when it has a tty.)
-
-supervised_virt_install () {
-expect -- - virt-install "$@" <<'EOF'
-eval spawn $argv
-set timeout -1
-expect {
-  "emergency mode" { send_user "\n\n\n\nABORT - emergency shell\n"
-                     exit 1
-                   }
-  "Please make your choice" { send_user "\n\n\n\nABORT - Anaconda stopped\n"
-                              exit 1
-                            }
-  "Please make a selection" { send_user "\n\n\n\nABORT - Anaconda stopped\n"
-                              exit 1
-                            }
-}
-EOF
-}
-
 cleanup_builder_vm
 
-supervised_virt_install $connect \
+virt-install $connect \
                         --name=$name \
-                        --extra-args="inst.stage2=hd:LABEL=RHEL-9-2-0-BaseOS-x86_64 inst.ks=file:/$fleet_ks_file console=ttyS0,115200" \
+                        --extra-args="inst.stage2=hd:LABEL=RHEL-${VERSION}-0-BaseOS-x86_64 inst.ks=file:/$fleet_ks_file console=ttyS0,115200" \
                         --disk "path=$out,format=qcow2" \
                         --ram 4096 \
                         --vcpus=1 \
                         --location="$url" \
                         --nographics \
                         --initrd-inject="$(realpath $fleet_ks_file)" \
-                        --noreboot
-rm "$fleet_ks_file"
+                        --noreboot \
+                        --serial=file,path="$(realpath $virt_log_file)" \
+                        --noautoconsole
+
+function get_state() {
+ virsh $connect dominfo $1 | grep State: | sed 's/State:\s*//'
+}
+
+# Wait until installation ends (with maximum timeout 30*20 sec)
+# check console output for errors if it appears to end early
+# if machine succeed it goes to "shut off" state.
```

Comment:
Isn't this what virt-install's `--wait` option is for? So `--wait 10` should wait for 10 mins, like you do here, but without shell polling.

Reply:
> Isn't this what virt-install's `--wait` option is for? So `--wait 10` should wait for 10 mins, like you do here, but without shell polling.

I think it could be used, but problem will be caused, when the error occurs, it will hang for 10 minutes. Instead of fail early. because fail is faster than installation.